cmake/modules/FindPythonLibsNew.cmake

```diff
@@ -170,6 +170,19 @@ else()
   # If all else fails, just set the name/version and let the linker figure out the path.
   if(NOT PYTHON_LIBRARY)
     set(PYTHON_LIBRARY python${PYTHON_LIBRARY_SUFFIX})
+    # Since this isn't very robust. One more try with find_libpython.py
+    execute_process(
+      COMMAND
+        "${PYTHON_EXECUTABLE}" "${CMAKE_SOURCE_DIR}/cmake/find_libpython.py"
```

Comment:
I haven't taken `find_libpython.py` for a spin, but looking at the copied-in implementation, I'm wondering if we're not adding some technical debt here. If it's absolutely needed for M1, maybe we could leverage modern CMake features (>3.12) , something like:
https://cmake.org/cmake/help/latest/module/FindPython.html
Haven't explored those either, so cannot say if we could rely on them. If yes, then I'd very much prefer adding a modern CMake dependency for M1 than adding extra code.

Reply:
@pramodk suggested a test to see if a late model FindPythonLibs (CMake 3.19.4 on the Apple M1) will make the ```find_libpython.py``` unnecessary. So I tried:
```
-  find_package(PythonLibsNew ${PYTHON_VERSION_MAJOR} REQUIRED)
+  find_package(PythonLibs REQUIRED)
```
This yields the potentially promising
```
--   LIB         | /Library/Developer/CommandLineTools/Library/Frameworks/python3.framework
```
but unfortunately, 'make' gives:
```
 => LINKING shared library ./libnrnmech.dylib
ld: can't map file, errno=22 file '/Library/Developer/CommandLineTools/Library/Frameworks/python3.framework' for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

With respect to "technical debt", I see this as a somewhat provisional solution which would not be modified by ourselves but replaced by something better as needed. Maybe someday Python itself will have a sys.shared_library_path(). Or cmake's PythonLibs module will become more robust. And ```find_libpython.py``` could end up reducing my currently very large ```nrnpyenv.sh``` technical debt by eliminating 90% of it.

Reply:
I see. Then no objections on my side.

Reply:
Ok. We can add this until we have robust cmake fix.